react/features/local-recording/recording/flac/flacEncodeWorker.js

```diff
@@ -0,0 +1,416 @@
+
+            this._data = this._exportFlacBlob();
+        }
+    }
+
+    /**
+     * Gets the stats.
+     *
+     * @returns {Object}
+     */
+    getStats() {
+        return {
+            'samplesEncoded': this._bufferSize
+        };
+    }
+
+    /**
+     * Gets the encoded flac file.
+     *
+     * @returns {Blob} - The encoded flac file.
+     */
+    getBlob() {
+        if (this._state === EncoderState.FINISHED) {
+            return this._data;
+        }
+
+        return null;
+    }
+
+    /**
+     * Converts flac buffer to a Blob.
+     *
+     * @private
+     * @returns {void}
+     */
+    _exportFlacBlob() {
+        const samples = mergeUint8Arrays(this._flacBuffers, this._flacLength);
+
+        const blob = new Blob([ samples ], { type: 'audio/flac' });
+
+        return blob;
+    }
+
+    /* eslint-disable no-unused-vars */
+    /**
+     * Callback function for saving encoded Flac data.
+     * This is invoked by libflac.
+     *
+     * @private
+     * @param {*} buffer - The encoded Flac data.
+     * @param {*} bytes - Number of bytes in the data.
+     * @returns {void}
+     */
+    _onEncodedData(buffer, bytes) {
+        this._flacBuffers.push(buffer);
+        this._flacLength += buffer.byteLength;
+    }
+    /* eslint-enable no-unused-vars */
+
+    /**
+     * Callback function for receiving metadata.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onMetadataAvailable = () => {
+        // reserved for future use
+    }
+}
+
+
+let encoder = null;
+
+self.onmessage = function(e) {
+
+    switch (e.data.command) {
+    case MAIN_THREAD_INIT:
+    {
+        const bps = e.data.config.bps;
+        const sampleRate = e.data.config.sampleRate;
+
+        if (Flac.isReady()) {
+            encoder = new Encoder(sampleRate, bps);
+            self.postMessage({
+                command: WORKER_LIBFLAC_READY
+            });
+        } else {
+            Flac.onready = function() {
+                setTimeout(() => {
+                    encoder = new Encoder(sampleRate, bps);
+                    self.postMessage({
+                        command: WORKER_LIBFLAC_READY
+                    });
+                }, 0);
+            };
+        }
+        break;
+    }
+
+    case MAIN_THREAD_NEW_DATA_ARRIVED:
+        if (encoder === null) {
+            console
+                .error('flacEncoderWorker:'
+                + 'received data when the encoder is not ready.');
+        } else {
+            encoder.encode(e.data.buf);
+        }
+        break;
+
+    case MAIN_THREAD_FINISH:
+        if (encoder !== null) {
+            encoder.finish();
+            const data = encoder.getBlob();
+
+            self.postMessage(
+                {
+                    command: WORKER_BLOB_READY,
+                    buf: data
+                }
+            );
+            encoder = null;
+        }
+        break;
+    }
+};
+
+/**
```

Comment:
What is this commented out chunk for?

Reply:
That was a piece of code for buffering the audio bits that arrived before libflac finishes initialization. Now with the `ensureInitialized` approach, this is handled in the adapter, so it is not necessary at the flacEncodeWorker level. I will remove it.